kernel/fablib/runlevel/0_infrastructure/aws_ssh_key/aws_ssh_key.go

```diff
@@ -0,0 +1,120 @@
+package aws_ssh_key
+
+import (
+	"crypto/x509"
+	"encoding/pem"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/openziti/fablab/kernel/model"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+	"golang.org/x/crypto/ssh"
+	"io/ioutil"
+	"path"
+)
+
+var KeyManager = awsKeyManager{}
+
+func Express() model.InfrastructureStage {
+	return KeyManager
+}
+
+type awsKeyManager struct {
+}
+
+func (l awsKeyManager) Bootstrap(m *model.Model) error {
+	bindings := model.GetBindings()
+
+	if !bindings.Has("credentials", "aws", "ssh_key_name") {
+		bindings.Put(bindings.Must("environment"), "credentials", "aws", "ssh_key_name")
```

Comment:
It _might_ also make more sense to use the instance id for this, rather than user data? I'm assuming `environment` is a new flag you're using in the binding?

Reply:
environment is an existing field in fablab.yml. instance id would make sense if you want to make it per-instance.

Reply:
Or maybe combine it? So you'd end up with something like plorenz-fablab-tiny (if my instance was named tiny)

Reply:
not the _AWS_ instance id... the `fablab` model instance id.

Reply:
`fablab create` ... generates an instance id: `~/.fablab/instances/<id>`

Reply:
`environment` is what all of the instances a user owns would get tagged. the `fablab` instance id would be a specific instance of a model... which is what would _seem_ to make sense as the ssh key identity (to me).

Reply:
`environment` would be fine, as long as this code is idempotent and will deal with an ssh key being present already... if i have 2 model instances, and they both want an ssh key named `environment`... will that work?

Reply:
(but considering the `dispose` hooks, I'm assuming it's not)